### PR TITLE
feat: submit "try osnap" form to airtable

### PR DESCRIPTION
**Pull request by ryanwolhuter**
When the user clicks "try osnap" show the "try osnap" modal
When the user submits the modal form, create a new record in the "Inbound Prospects" table in Airtable

Using the Airtable API requires an api key, and as per their docs "Using Airtable.js in the browser is strongly discouraged, since that will expose your API key to everyone who has access to that web page."

To get around this I have added an API route that we POST the form data to. Nextjs then handles the Airtable code server-side and therefore does not need a "NEXT_PUBLIC_" type env var for the api key.

You will also notice that I have used the ID strings for the table and fields as per the docs' recommendation. This means that if they change the names of the table or fields this code will still work.

**NOTE:**

Since this code needs a nextjs api route, we would not be able to test it on Storybook. I have therefore added the whole new page to the /osnap-v2 route. Testing can be done there.

### CODE REVIEW

File: src/app/api/airtable/route.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const apiKey = process.env.AIRTABLE_API_KEY!;
+const baseId = "appU5MIM7Yl1VrxEm";
+const tableId = "tbl2scUK2gyVCSWtj";
+const organizationFieldId = "fldyCZtTyHT8jWLXa";
+const integrationFieldId = "fldafGL8jvE3toWk8";
+const nameFieldId = "fldX9QMw47XvzSW86";
+const communicationChannelFieldId = "fldjxMaR5DEtF29GC";
+const contactDetailsFieldId = "fldRByHAhU5Wbg9pF";
+console.log({ apiKey });
```
Comment (daywiss) [inline src/app/api/airtable/route.ts:14]: stray log

File: src/app/api/airtable/route.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  try {
+    const record = await airtableBase(tableId).create(fields);
+    const recordId = record.getId();
+    console.log(`Successfully created record ${recordId} in table ${tableId}`);
+    return NextResponse.json({
+      status: 200,
+      body: `Successfully created record ${recordId} in table ${tableId}`,
+    });
+  } catch (error) {
```
Comment (daywiss) [inline src/app/api/airtable/route.ts:39]: surprised next doesnt catch this error automatically to pass back to caller

File: src/app/api/airtable/route.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    console.error("Failed to create record in Airtable", error);
+    return NextResponse.json({
+      status: 500,
+      body: `Failed to create record in Airtable`,
```
Comment (daywiss) [inline src/app/api/airtable/route.ts:43]: normally i would do something like `Failed to create record in Airtable: ${error.message}`.  i see you do log it, so its ok if thats easy to see in production

**Review by daywiss — APPROVED** [review]
nice work, and good use of backend route

File: src/app/api/airtable/route.ts (hunk 2 above)
Comment (ryanwolhuter) [reply src/app/api/airtable/route.ts:39]: yeah and I discovered this the hard way

**Pull request merged**